milner_desvars/wk3-notes.md

```diff
@@ -0,0 +1,313 @@
+# Week 3 Notes
+
+# Master
+
+## How to run master node from terminal
+
+The command to run the master node from a terminal is:
+
+`roscore`
+
+## rosrun
+
+
+rosrun is a command from the rosbash package that allows you to run a single node from a package without needing to give it's full path or cd/roscd there first.
+
+Usage 
+
+`rosrun <package> <executable>`
+
+Ex.
+
+`rosrun gazebo_ros gazebo`
+
+## roslaunch
+
+roslaunch is a package that allows you to run multiple nodes locally and remotely via SSH, as well as setting on the Parameter Server.
+
+It takes one or more XML configuration files (with the .launch extension) that specify the parameters to set and the nodes to launch, as well as the machines that they should be run on.
+
+Ex. 
+
+`roslaunch ca_gazebo create_house.launch`
+
+## Difference between rosrun and roslaunch commands
+
+The main difference is that rosrun can only run one node at the time, and also need the master to be initialized (with `roscore`).
+
+roslaunch, on the other hand, takes an XML file that has all the information about what nodes to launch, so it can launch multiples nodes at once and also automatically runs the master if it is not already running.
+
+## ROS Architecture
+
+### ROS Filesystem Level
+
+Mainly covers ROS resources that you encounter in the disk, such as:
+
+* Packages
+* Metapackages
+* Package Manifiest
+* Repositories
+* Message (msg) types
+* Service (srv) types
+
+### ROS Computation Graph Level
+
+Are the peer-to-peer network of ROS processes that are processing data together, such as:
+
+* Nodes
+* Master
+* Parameter Server
+* Messages
+* Topics
+* Services
+* Bags
+
+### ROS Community Level
+
+Are the ROS Resources that enable separate communities to exchange software and knowledge, such as:
+
+* Distributions
+* Repositories
+* The ROS Wiki
+* Bug Ticket System
+* Mailing Lists
+* ROS Answers
+* Blog
+
+## How launch files are organized in the project?
+
+### create_autonomy
+
+In the project create_autonomy, the launch files are organized in the packages in which they are implemented, so If a package needs to include another launch file in their XML configuration, it uses a command to find that package path to do a proper call to the launch file that would be added.
+
+### turtlebot
+
+In the project turtlebot, also the launch files are organized in the respective packages, the difference resides in that, if a launch file need to include another one, the included launch file  is on a folder called "include" in the same folder as the original launch file, also the format of the file is `.launch.xml`, different from the `.launch` original.
+
+## Profiling and debugging tools?
+
+### Valgrind
+
+The Valgrind tool suite provides several debugging and profiling tools that help you make your programs faster and more correct.
```

Comment:
I don't like this definition because it doesn't reflect its usage. Check [this one](https://en.wikipedia.org/wiki/Valgrind).